src/graph.erl

```diff
@@ -297,8 +409,66 @@ list_parents(Vertex) -> persistence:set_list_members(parents_id(Vertex)).
 -spec list_children(Vertex :: binary()) -> {ok, list(binary())} | {error, any()}.
 list_children(Vertex) -> persistence:set_list_members(children_id(Vertex)).
 
-
 %% @todo implement this - basically all zones that this zone has contact with
 -spec all_zones() -> {ok, list(binary())} | {error, any()}.
 all_zones() ->
     {ok, []}.
+
+-spec effective_list_parents(Vertex :: binary()) -> list(binary()).
+effective_list_parents(Vertex) -> do_list(persistence:set_list_members(effective_parents_id(Vertex))).
+
+do_list({ok, Result}) -> Result;
+do_list({error, _Error}) -> [].
```

Comment:
This is incompatible with every other function in this module. Why did you do this like that?

Reply:
In optimistic scenario, which we do plenty of here, {error, Reason} indicates that the entry doesn't exist. This function is used for effective children and parents and if there isn't an entry in database, there are no effective children and parents

Reply:
I suppose. It seems the better option is that database returns an error (not an empty result) then naive query should be called.

I will accept it for now as I want to run tests on k8s, but you should refactor it asap.